### PR TITLE
user roles & admin functionality

### PR DESCRIPTION
**Pull request by sethvincent**
This pr adds user roles to the project.

It also sets up authentication mocking so we can run tests that involve logged in users. I used [passport-mock-strategy](https://github.com/cszatma/passport-mock-strategy) for this. We may need to add to the mock user that is created depending on the other testing needs that come up.

todo before this is ready:
- [x] fix running tests on CI
- [x] lint fixes
- [x] fix api mistakes
- [x] update changelog
- [ ] some ui implementation

### CODE REVIEW

File: api/test.sh [hunk 1]
```diff
@@ -1,7 +1,9 @@
 if [ -z "$CI" ]; then
-    createdb -p 5433 scoreboard_tests && 
-    ava tests/*.test.js ; 
-    dropdb  -p 5433 scoreboard_tests
+    docker-compose exec db createdb -U postgres scoreboard_tests
```
Comment (sethvincent) [inline api/test.sh]: I had a hard time getting tests to run and made these changes.

Not sure if there are any issues here, but this is working well locally.

I need to revise the `else` part of this statement to work on CI.

File: api/bin/commands/role/set.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    return process.exit(1)
+  }
+
+  if (!role) {
+    console.error('role is required')
+    return process.exit(1)
+  }
+
+  const users = new Users()
+  console.log('users', users)
```
Comment (kamicut) [inline api/bin/commands/role/set.js]: Remove this console.log?

File: api/src/models/users.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 const connection = require('../db/connection')
+const { validateRole } = require('../utils/roles')
 
 class Users {
```
Comment (kamicut) [inline api/src/models/users.js]: To add a role, do we just call `update`? Do we want to make it more explicit than that?

**Comment by kamicut** [comment]
@sethvincent this PR is affected by the merge @alyssadelaine has made in another branch. Let me know if you need help there. Also, do you want to move UI additions to another PR? I am wondering if we need a UI so that an admin sets roles for other users

**Pull request merged**